Version 7.1.0 seems like not picking up conftest.py

Over SciPy we have started to see the message in the title in our Azure Pipelines which started picking up 7.1.0 instead of 7.0.1 lastn ight. From the deprecation and removal list we couldn't see which item might have caused it. 

One speculation we might generate is that our `slow` mark is not registered in our `pytest.ini` file explicitly and maybe registrations are now obligatory. 

Here is a complete SciPy test suite example with all installation steps provided. 

https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=16988&view=logs&jobId=7d84eb14-5b07-5555-8e62-838a7038a2b0&j=ede89041-5e8e-5201-ce5c-f68c2cbf02dd&t=141d9d53-a127-58b4-0262-4b9693c8e91f

And our pytest.ini file lives here 

https://github.com/scipy/scipy/blob/main/pytest.ini

Thank you for all the efforts.

We also noticed an unexpected change in behavior in our CI with v7.1.0 that looks related to this.

We're calling `pytest --pyargs module_name` where `module_name` is an installed module and at the same time in the current working directory there is a file that contains a pytest config block with a list of markers.

With v7.1.0, it seems like the fixtures in the installed `module_name.tests.conftest` are not loaded when the local settings are present. If we delete the pytest config block from the local file then the fixtures are loaded and the test suite can run without errors (but with warnings about unknown markers).

It doesn't seem to matter whether the config block is in `setup.cfg` or `pyproject.toml`.

I don't have a proper minimal example at this point, but you can replicate this with:

```
pip install -r https://raw.githubusercontent.com/explosion/thinc/v8.0.14/requirements.txt
pip install thinc==8.0.14
pip install pytest==7.1.0
wget https://raw.githubusercontent.com/explosion/thinc/v8.0.14/setup.cfg
pytest --pyargs thinc.tests.test_config::test_config_roundtrip_disk_respects_path_subclasses
```

The error is a missing fixture:

```
  def test_config_roundtrip_disk_respects_path_subclasses(pathy_fixture):
E       fixture 'pathy_fixture' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.
```

`pathy_fixture` is defined in `thinc.tests.conftest`: https://github.com/explosion/thinc/blob/v8.0.14/thinc/tests/conftest.py

If you downgrade to `pytest==7.0.1` there are no errors, or if you delete `setup.cfg` with v7.1.0 there are potentially marker warnings but no errors.

We've always had a bit of trouble with the options and marker definitions in `conftest.py` when running the test suite from within an installed module vs. a local directory, so it's possible there's another interaction there. The goal is to add a `--slow` option, but we've never gotten it to work 100% as intended for installed modules. We've mainly been using the local ini file to get rid of the marker warnings in the CI.

There didn't seem to be anything specific to OS or python versions in the CI errors we've seen so far.

<details><summary>pip list for reference:</summary>

```
Package             Version
------------------- -------
asttokens           2.0.5  
attrs               21.4.0 
backcall            0.2.0  
bleach              4.1.0  
blis                0.7.6  
catalogue           2.0.6  
click               8.0.4  
coverage            5.5    
cymem               2.0.6  
Cython              0.29.28
decorator           5.1.1  
defusedxml          0.7.1  
entrypoints         0.4    
executing           0.8.3  
flake8              3.5.0  
hypothesis          6.39.3 
importlib-resources 5.4.0  
iniconfig           1.1.1  
ipykernel           5.1.4  
ipython             8.1.1  
ipython-genutils    0.2.0  
jedi                0.18.1 
Jinja2              3.0.3  
jsonschema          4.4.0  
jupyter-client      7.1.2  
jupyter-core        4.9.2  
jupyterlab-pygments 0.1.2  
MarkupSafe          2.1.0  
matplotlib-inline   0.1.3  
mccabe              0.6.1  
mistune             0.8.4  
ml-datasets         0.2.0  
mock                2.0.0  
murmurhash          1.0.6  
mypy                0.910  
mypy-extensions     0.4.3  
nbclient            0.5.13 
nbconvert           6.1.0  
nbformat            5.1.3  
nest-asyncio        1.5.4  
numpy               1.22.3 
packaging           21.3   
pandocfilters       1.5.0  
parso               0.8.3  
pathy               0.6.1  
pbr                 5.8.1  
pexpect             4.8.0  
pickleshare         0.7.5  
pip                 20.0.2 
pkg-resources       0.0.0  
pluggy              1.0.0  
preshed             3.0.6  
prompt-toolkit      3.0.28 
ptyprocess          0.7.0  
pure-eval           0.2.2  
py                  1.11.0 
pycodestyle         2.3.1  
pydantic            1.8.2  
pyflakes            1.6.0  
Pygments            2.11.2 
pyparsing           3.0.7  
pyrsistent          0.18.1 
pytest              7.1.0  
pytest-cov          2.7.1  
python-dateutil     2.8.2  
pyzmq               22.3.0 
setuptools          60.9.3 
six                 1.16.0 
smart-open          5.2.1  
sortedcontainers    2.4.0  
srsly               2.4.2  
stack-data          0.2.0  
testpath            0.6.0  
thinc               8.0.14 
toml                0.10.2 
tomli               2.0.1  
tornado             6.1    
tqdm                4.63.0 
traitlets           5.1.1  
typer               0.4.0  
types-mock          4.0.11 
typing-extensions   4.1.1  
wasabi              0.9.0  
wcwidth             0.2.5  
webencodings        0.5.1  
wheel               0.37.1 
zipp                3.7.0
```

</details>

Here is some additional debugging:

```python
    def _is_in_confcutdir(self, path: Path) -> bool:
        """Whether a path is within the confcutdir.

        When false, should not load conftest.
        """
        if self._confcutdir is None:
            return True
        oldret = path not in self._confcutdir.parents
        try:
            path.relative_to(self._confcutdir)
        except ValueError as e:
            if oldret:
                print(f"\nOLD True NEW False:\n  {e}\n  {self._confcutdir=}\n  {path=})")
            return False
        if not oldret:
            print(f"\nOLD False NEW True:\n  {self._confcutdir=}\n  {path=}")
        return True
```

resulting output with the reproducer above:

```
─[florian@aragog 1]──[~/tmp/9767-2/thinc]──[22-03-16]──[18:02]──[git/tags/v8.0.14•]────┄
$ ../.venv/bin/python3 -m pytest -s --pyargs thinc.tests.test_config::test_config_roundtrip_disk_respects_path_subclasses
================================================ test session starts =================================================
platform linux -- Python 3.10.2, pytest-7.1.0.dev251+g4eaa6aee7, pluggy-1.0.0
rootdir: /home/florian/tmp/9767-2/thinc, configfile: setup.cfg
plugins: hypothesis-6.39.3, cov-2.7.1
collecting ... 
OLD True NEW False:
  '/home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages/thinc/tests/test_config.py' is not in the subpath of '/home/florian/tmp/9767-2/thinc' OR one path is relative and the other is absolute.
  self._confcutdir=PosixPath('/home/florian/tmp/9767-2/thinc')
  path=PosixPath('/home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages/thinc/tests/test_config.py'))

OLD True NEW False:
  '/home/florian/tmp/9767-2/.venv' is not in the subpath of '/home/florian/tmp/9767-2/thinc' OR one path is relative and the other is absolute.
  self._confcutdir=PosixPath('/home/florian/tmp/9767-2/thinc')
  path=PosixPath('/home/florian/tmp/9767-2/.venv'))

OLD True NEW False:
  '/home/florian/tmp/9767-2/.venv/lib' is not in the subpath of '/home/florian/tmp/9767-2/thinc' OR one path is relative and the other is absolute.
  self._confcutdir=PosixPath('/home/florian/tmp/9767-2/thinc')
  path=PosixPath('/home/florian/tmp/9767-2/.venv/lib'))

OLD True NEW False:
  '/home/florian/tmp/9767-2/.venv/lib/python3.10' is not in the subpath of '/home/florian/tmp/9767-2/thinc' OR one path is relative and the other is absolute.
  self._confcutdir=PosixPath('/home/florian/tmp/9767-2/thinc')
  path=PosixPath('/home/florian/tmp/9767-2/.venv/lib/python3.10'))

OLD True NEW False:
  '/home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages' is not in the subpath of '/home/florian/tmp/9767-2/thinc' OR one path is relative and the other is absolute.
  self._confcutdir=PosixPath('/home/florian/tmp/9767-2/thinc')
  path=PosixPath('/home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages'))

OLD True NEW False:
  '/home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages/thinc' is not in the subpath of '/home/florian/tmp/9767-2/thinc' OR one path is relative and the other is absolute.
  self._confcutdir=PosixPath('/home/florian/tmp/9767-2/thinc')
  path=PosixPath('/home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages/thinc'))

OLD True NEW False:
  '/home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages/thinc/tests' is not in the subpath of '/home/florian/tmp/9767-2/thinc' OR one path is relative and the other is absolute.
  self._confcutdir=PosixPath('/home/florian/tmp/9767-2/thinc')
  path=PosixPath('/home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages/thinc/tests'))
collected 1 item                                                                                                     

. E

======================================================= ERRORS =======================================================
_______________________ ERROR at setup of test_config_roundtrip_disk_respects_path_subclasses ________________________
file /home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages/thinc/tests/test_config.py, line 352
  def test_config_roundtrip_disk_respects_path_subclasses(pathy_fixture):
E       fixture 'pathy_fixture' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/florian/tmp/9767-2/.venv/lib/python3.10/site-packages/thinc/tests/test_config.py:352
============================================== short test summary info ===============================================
ERROR ::test_config_roundtrip_disk_respects_path_subclasses
================================================== 1 error in 0.26s ==================================================
```

Gonna stop here because I'm somewhat tired and getting more confused the longer I look at this issue to be honest...

--- a/changelog/9767.bugfix.rst
+++ b/changelog/9767.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed a regression in pytest 7.1.0 where some conftest.py files outside of the source tree (e.g. in the `site-packages` directory) were not picked up.

--- a/src/_pytest/config/__init__.py
+++ b/src/_pytest/config/__init__.py
@@ -540,11 +540,7 @@ class PytestPluginManager(PluginManager):
         """
         if self._confcutdir is None:
             return True
-        try:
-            path.relative_to(self._confcutdir)
-        except ValueError:
-            return False
-        return True
+        return path not in self._confcutdir.parents
 
     def _try_load_conftest(
         self, anchor: Path, importmode: Union[str, ImportMode], rootpath: Path

--- a/testing/test_conftest.py
+++ b/testing/test_conftest.py
@@ -252,6 +252,34 @@ def test_conftest_confcutdir(pytester: Pytester) -> None:
     result.stdout.no_fnmatch_line("*warning: could not load initial*")
 
 
+def test_installed_conftest_is_picked_up(pytester: Pytester, tmp_path: Path) -> None:
+    """When using `--pyargs` to run tests in an installed packages (located e.g.
+    in a site-packages in the PYTHONPATH), conftest files in there are picked
+    up.
+
+    Regression test for #9767.
+    """
+    # pytester dir - the source tree.
+    # tmp_path - the simulated site-packages dir (not in source tree).
+
+    pytester.syspathinsert(tmp_path)
+    pytester.makepyprojecttoml("[tool.pytest.ini_options]")
+    tmp_path.joinpath("foo").mkdir()
+    tmp_path.joinpath("foo", "__init__.py").touch()
+    tmp_path.joinpath("foo", "conftest.py").write_text(
+        textwrap.dedent(
+            """\
+            import pytest
+            @pytest.fixture
+            def fix(): return None
+            """
+        )
+    )
+    tmp_path.joinpath("foo", "test_it.py").write_text("def test_it(fix): pass")
+    result = pytester.runpytest("--pyargs", "foo")
+    assert result.ret == 0
+
+
 def test_conftest_symlink(pytester: Pytester) -> None:
     """`conftest.py` discovery follows normal path resolution and does not resolve symlinks."""
     # Structure:

